src/account/accountsReducer.js

```diff
@@ -110,6 +112,34 @@ export default (state: AccountsState = initialState, action: Action): AccountsSt
     case ACCOUNT_REMOVE:
       return accountRemove(state, action);
 
+    case EVENT: {
+      const { event } = action;
+      switch (event.type) {
+        case 'restart': {
+          const { zulip_feature_level, zulip_version } = event;
+          if (zulip_feature_level !== undefined && zulip_version !== undefined) {
+            // TODO (?): Detect if these are different from the values
+            // we had before, so we know it's an upgrade, not just a
+            // restart. Then, implement logic like "be sure to refetch
+            // from scratch within N hours" (to avoid thundering
+            // herding the server).
+            return [
+              {
+                ...state[0],
```

Comment:
Will the active account always be `state[0]`? This seems like a brittle way of doing things in general, although possibly it's too embedded to easily change at this point. (This isn't really a comment on this PR, more of a general question)

Reply:
> Will the active account always be `state[0]`

Yeah, that's how it is. There's been some discussion on the brittleness of that; I don't remember where.